Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1514,6 +1532,191 @@ DiscusMultipleScatteringCorrection::scatter(const int nScatters, Kernel::PseudoR
   return {true, weights};
 }
 
+/*
+ * Construct a hexahedron shape extending from the detector's front face across the collimator openning window toward
+ the sample with a legth as twice the distance from sample to detector.
+ */
+const std::shared_ptr<Geometry::CSGObject> DiscusMultipleScatteringCorrection::createCollimatorHexahedronShape(
+    const Kernel::V3D &samplePos, const Mantid::Geometry::DetectorInfo &detectorInfo, const size_t &histogramIndex) {
+  const auto shape = detectorInfo.detector(histogramIndex).shape();
+  if (!shape || (shape->shape() != Mantid::Geometry::detail::ShapeInfo::GeometryShape::CUBOID)) {
+    return nullptr;
+  }
+
+  try {
+    shape->shapeInfo();
+  } catch (std::exception &) {
+    return nullptr;
+  }
+
+  const auto colCorridorShape = readFromCollimatorCorridorCache(histogramIndex);
+  if (colCorridorShape) {
+    return colCorridorShape;
+  }
+
+  const auto &detectorId = detectorInfo.detector(histogramIndex).getID();
+  const auto &detectorAbsolPos = detectorInfo.position(detectorInfo.indexOf(detectorId));
+  const auto cuboidGeometry = shape->shapeInfo().cuboidGeometry();
+
+  // Positions of the detector's front face
+  const auto &detLeftFrontBottomPos = detectorAbsolPos + cuboidGeometry.leftFrontBottom;
+  const auto &detLeftFrontTopPos = detectorAbsolPos + cuboidGeometry.leftFrontTop;
+  const auto &detRightFrontBottomPos = detectorAbsolPos + cuboidGeometry.rightFrontBottom;
+  const auto detRightFrontTopPos = detLeftFrontTopPos + detRightFrontBottomPos - detLeftFrontBottomPos;
+
+  const auto detCentrePos =
+      (detLeftFrontBottomPos + detLeftFrontTopPos + detRightFrontBottomPos + detRightFrontTopPos) / 4.0;
+  const auto detTopMiddlePos = (detLeftFrontTopPos + detRightFrontTopPos) / 2.0;
+
+  // Sanity check to avoid dividing by zero
+  if ((detRightFrontTopPos == detLeftFrontTopPos) || (detTopMiddlePos == detCentrePos) || (detCentrePos == samplePos)) {
+    return nullptr;
+  }
+
+  // Define the unit vectors needed for calculations
+  const auto unitVecSampleToDet = Kernel::normalize(detCentrePos - samplePos);
+  const auto unitVecLeftToRight = Kernel::normalize(
+      V3D(-1.0 * unitVecSampleToDet.Z(), 0,
+          -1.0 * unitVecSampleToDet.X())); // this is a vector normal to unitVecSampleToDet on XZ plane
```

Comment:
This works for getting the vector in the scattering plane on ENGIN-X (and WISH - another instrument with a radial collimator)I think - but it won't work for all instruments. In future perhaps you can get the ki vector from the workspace, however I'm happy to leave it as is for now - want to get this in the release (my fault for dragging my feet with the testing sorry)!